src/audit/secrets_outside_environment.rs

```diff
@@ -0,0 +1,80 @@
+use github_actions_models::{
+    common::{expr::LoE, Env, EnvValue},
+    workflow::job::StepBody,
+};
+
+use super::{audit_meta, Audit};
+use crate::{finding::Confidence, models::Job};
+
+pub(crate) struct SecretsOutsideEnvironment;
+
+audit_meta!(
+    SecretsOutsideEnvironment,
+    "secrets-outside-environment",
+    "secrets used without an environment to gate them"
+);
+
+impl Audit for SecretsOutsideEnvironment {
+    fn new(_state: super::AuditState) -> anyhow::Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self)
+    }
+
+    fn audit_raw<'w>(
```

Comment:
Unless I'm missing something, I don't think you want `audit_raw` here -- that's only for low-level audits like `unredacted-secrets`.

Since you're operating purely on the YAML here and are focused on step-level findings, I think you want `audit_step` here.

Reply:
Or rather, `audit_normal_job`, since you're testing the job's `environment`.

Reply:
Thank you! I went for the `audit_step` and access the `environment` by going up the hierarchy. Addressed in `1d5b2df`.